### DISCUSSION TITLE
Is it ok to proxy the Keyclocks APIs

### DISCUSSION THREAD

**Discussion by iamamirsalehi** [keycloak/keycloak#28230] [Q&A]
Hi there

Our website needs to have popup models for login and register and as I know Keycloak requires us to open a new tab for user to get logged in, so the solution came to my mind was to proxy the apis to have our custom login - register scenario, Is it technically ok? or is there any better solution?

**Comment by codespearhead** [comment]
- Is it acceptable?

Yes.

- Is it recommended?

Not at all. What will you do when a different client, such as another site, wants to access the login form? Will you replicate the entire login markup and logic in the other site? They'll eventually go out of sync. Will you redirect the second site to the first site? Then you're reinventing the Keycloak Authorization server with less security harnesses. That's one of the reasons OAuth2 was invented btw. I suggest creating a custom Keycloak theme.
